File: 2020/3/1.py
input = open("input", "r").read().split('\n')

DOWN=1
RIGHT=3

i = RIGHT
j = DOWN
maxIndex = len(input[0])

trees = 0
while j <= len(input) - 1:
  if input[j][i % maxIndex] == '#':
    trees += 1

  i += RIGHT
  j += DOWN

print('NUMBER OF TREES: ' + str(trees))

File: 2022/1/1.ts
import { readInput } from "../../read-input";

export const solve1 = (inputPath: string): number => {
  const data = readInput(inputPath);
  const elves = data.split("\n\n").map((x) => x.split("\n"));

  const sums = elves.map((x) => x.reduce((a, b) => a + Number(b), 0));

  sums.sort((a, b) => b - a);

  return sums[0];
};

export const solve2 = (inputPath: string): number => {
  const data = readInput(inputPath);
  const elves = data.split("\n\n").map((x) => x.split("\n"));

  const sums = elves.map((x) => x.reduce((a, b) => a + Number(b), 0));

  sums.sort((a, b) => b - a);

  return sums[0] + sums[1] + sums[2];
};
